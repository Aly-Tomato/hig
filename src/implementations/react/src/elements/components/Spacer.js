import React from "react";
import SpacerAdapter from "../../adapters/SpacerAdapter";

class Spacer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  setSlotEl = el => {
    this.setState({ setSlotEl: el });
  };

  render() {
    return (
      <SpacerAdapter {...this.props}>
        <div ref={this.setSlotEl}>{this.props.children}</div>
      </SpacerAdapter>
    );
  }
}

export default Spacer;
