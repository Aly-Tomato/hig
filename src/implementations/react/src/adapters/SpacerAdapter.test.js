import {mount} from 'enzyme';
import * as HIG from 'hig-vanilla';
import React from 'react';

import Slot from '../elements/components/GlobalNav/Slot';
import Spacer from './SpacerAdapter'
import SharedExamples from '../elements/components/GlobalNav/SharedExamples'

const Context = props => {
  return (
    <Spacer
      type={props.type}
      width={props.width}
      inset={props.inset}
    />
  );
};

function setupReactContext(props) {
  const reactContainer = document.createElement("div");
  const reactWrapper = mount(<Context {...props}/>,
  {
    attachTo: reactContainer
  });
  return {reactWrapper, reactContainer}
}

function createHigContext(props) {
  const higContainer = document.createElement('div');

  const higSpacer = new HIG.Spacer({...props});
  higSpacer.mount(higContainer);

  return {higContainer, higItem: higSpacer}
}

describe("Spacer", () => {
  it('renders the basic spacer', () => {
    const props = {};
    const {reactWrapper, reactContainer} = setupReactContext(props);
    const {higContainer, higItem} = createHigContext(props);

    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
    expect(reactContainer.firstElementChild.outerHTML).toEqual(higContainer.firstElementChild.outerHTML);
  });

  it('renders some non defaults', () => {
    const props = {
      type: 'stack', width: 'none', inset: 'xxl'
    };

    const {reactWrapper, reactContainer} = setupReactContext(props);
    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
  });

  describe('setting and updating props', () => {
    const shex = new SharedExamples(Context, createHigContext);

    const configSets = [{
      key: 'type', sampleValue: 'inline', updateValue: 'stack', mutator: 'setType'
    }, {
      key: 'width', sampleValue: 'xs', updateValue: 'xl', mutator: 'setWidth'
    }, {
      key: 'inset', sampleValue: 'xl', updateValue: 'none', mutator: 'setInset'
    }];

    configSets.forEach(function(config) {

      it(`can set props for ${config.key}`, () => {
        shex.verifyPropsSet(config);
      });

      it(`can update props for ${config.key}`, () => {
        shex.verifyPropsUpdate(config);
      });
    });
  });
});