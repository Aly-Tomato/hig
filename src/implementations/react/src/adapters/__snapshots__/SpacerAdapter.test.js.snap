// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer renders some non defaults 1`] = `"<span class=\\"hig__spacer hig__spacer--stack hig__spacer--width-none hig__spacer--inset-xxl\\"></span>"`;

exports[`Spacer renders the basic spacer 1`] = `"<span class=\\"hig__spacer hig__spacer--stack hig__spacer--width-m\\"></span>"`;

exports[`Spacer setting and updating props can set props for inset 1`] = `"<span class=\\"hig__spacer hig__spacer--stack hig__spacer--width-m hig__spacer--inset-xl\\"></span>"`;

exports[`Spacer setting and updating props can set props for type 1`] = `"<span class=\\"hig__spacer hig__spacer--inline hig__spacer--width-m\\"></span>"`;

exports[`Spacer setting and updating props can set props for width 1`] = `"<span class=\\"hig__spacer hig__spacer--stack hig__spacer--width-xs\\"></span>"`;

exports[`Spacer setting and updating props can update props for inset 1`] = `"<span class=\\"hig__spacer hig__spacer--stack hig__spacer--width-m hig__spacer--inset-none\\"></span>"`;

exports[`Spacer setting and updating props can update props for type 1`] = `"<span class=\\"hig__spacer hig__spacer--width-m hig__spacer--stack\\"></span>"`;

exports[`Spacer setting and updating props can update props for width 1`] = `"<span class=\\"hig__spacer hig__spacer--stack hig__spacer--width-xl\\"></span>"`;
